db/src/types.rs

```diff
@@ -38,12 +36,13 @@ pub enum ValueType {
 
 /// Represents a Mentat value in a particular value set.
 // TODO: expand to include :db.type/{instant,url,uuid}.
-#[derive(Clone,Debug,Eq,Hash,Ord,PartialOrd,PartialEq)]
+// Floats and doubles can't implement Eq and Hash.
+#[derive(Clone,Debug,PartialOrd,PartialEq)]
```

Comment:
We needed `Hash` so we're able to put `edn::Value` instances into a map. That's why we use `OrderedFloat` in the first place. I'm not sure if we'll need to use `TypedValue`s as keys in a map. What do you think, @ncalexan?